cdap-watchdog/src/main/java/co/cask/cdap/logging/plugins/FixedWindowRollingPolicy.java

```diff
@@ -0,0 +1,151 @@
+/*
+ * Copyright © 2017 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.logging.plugins;
+
+import ch.qos.logback.core.CoreConstants;
+import ch.qos.logback.core.rolling.RolloverFailure;
+import ch.qos.logback.core.rolling.helper.FileNamePattern;
+import ch.qos.logback.core.rolling.helper.IntegerTokenConverter;
+import co.cask.cdap.common.io.Locations;
+import org.apache.twill.filesystem.Location;
+
+import java.io.IOException;
+
+/**
+ * When rolling over, renames files according to a fixed window algorithm.
+ */
+public class FixedWindowRollingPolicy extends LocationRollingPolicyBase {
+  // TODO Support compression
+  private int minIndex;
+  private static final String FNP_NOT_SET =
+    "The \"FileNamePattern\" property must be set before using FixedWindowRollingPolicy. ";
+  private int maxIndex;
+
+  /**
+   * It's almost always a bad idea to have a large window size, say over 12.
+   */
+  private static final int MAX_WINDOW_SIZE = 12;
+
+  public FixedWindowRollingPolicy() {
+    minIndex = 1;
+    maxIndex = 7;
+  }
+
+  @Override
+  public void start() {
+    if (fileNamePatternStr != null) {
+      fileNamePattern = new FileNamePattern(fileNamePatternStr, this.context);
+    } else {
+      addError(FNP_NOT_SET);
+      addError(CoreConstants.SEE_FNP_NOT_SET);
+      throw new IllegalStateException(FNP_NOT_SET + CoreConstants.SEE_FNP_NOT_SET);
+    }
+
+    if (maxIndex < minIndex) {
+      addWarn("MaxIndex (" + maxIndex + ") cannot be smaller than MinIndex ("
+                + minIndex + ").");
+      addWarn("Setting maxIndex to equal minIndex.");
+      maxIndex = minIndex;
+    }
+
+    if ((maxIndex - minIndex) > MAX_WINDOW_SIZE) {
```

Comment:
It is hard coded to be `12` even in logback implementation. I changed it to `20` as per https://logback.qos.ch/xref/ch/qos/logback/core/rolling/FixedWindowRollingPolicy.html. They do not encourage using large window sizes